src/IO/Observer/ReductionActions.hpp

```diff
@@ -334,5 +391,161 @@ struct WriteReductionData {
     }
   }
 };
+
+/*!
+ * \ingroup ObserversGroup
+ * \brief Collect the reduction data from the Observer group on the
+ * ObserverWriter nodegroup before sending to node 0 for printing to screen.
+ */
+struct PrintReductionData {
+ private:
+  static void append_to_reduction_data(
+      const gsl::not_null<std::vector<std::string>*> all_reduction_data,
+      const std::string str) noexcept {
+    all_reduction_data->push_back(str);
+  }
+
+  static void append_to_reduction_data(
+      const gsl::not_null<std::vector<std::string>*> all_reduction_data,
+      const std::vector<std::string>& str_vec) noexcept {
+    all_reduction_data->insert(all_reduction_data->end(), str_vec.begin(),
+                               str_vec.end());
+  }
+
+  template <typename... Ts, size_t... Is>
+  static void print_data(std::tuple<Ts...>&& data,
+                         std::index_sequence<Is...> /*meta*/) noexcept {
+    static_assert(sizeof...(Ts) > 0,
+                  "Must be reducing at least one piece of data");
```

Comment:
Please make the error message more specific, e.g. "Must be reducing at least one piece of data in print_data()"